backend/src/email/middlewares/email-template.middleware.ts

```diff
@@ -379,10 +387,111 @@ const uploadProtectedCompleteHandler = async (
   }
 }
 
+/**
+ * Downloads the file from the vault url and checks that its columns match the attributes provided in the template.
+ * If a template has not yet been uploaded, do not write to the message logs, but prompt the user to upload a template first.
+ * If the template and csv do not match, prompt the user to upload a new file.
+ * @param req
+ * @param res
+ * @param next
+ */
+const tesseractHandler = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<Response | void> => {
+  const { campaignId } = req.params
+  if (ProtectedService.isProtectedCampaign(+campaignId)) {
```

Comment:
```suggestion
  if (await ProtectedService.isProtectedCampaign(+campaignId)) {
```
fix: `isProtectedCampaign` returns a `Promise` so we have to await it here.